### ISSUE TITLE
Client v1.14.0 floods STDOUT with mashie warnings

### ISSUE THREAD

**Issue by bentheax** [zendesk/zendesk_api_client_rb#323]
## Environment

Ruby 2.4.0
Hashie 3.5.5

This likely occurs as a result of using Hashie 3.5+ , which introduced the warning "feature".
 
## Reproducing

1. Call `client.tickets.last` (or anything, really)
2. Observe warning spam like the following:

```
You are setting a key that conflicts with a built-in method Hashie::Mash#class defined in
 Kernel. This can cause unexpected behavior when accessing the key via as a property. You
 can still access the key via the #[] method.
W, [2017-02-27T01:20:01.552016 #367]  WARN -- : You are setting a key that conflicts with
 a built-in method Hashie::Mash#class defined in Kernel. This can cause unexpected behavi
or when accessing the key via as a property. You can still access the key via the #[] met
hod.
```

## Expected behaviour

If the warnings are expected, silence them. See the section relating to `disable_warnings` in https://github.com/intridea/hashie .

## Workarounds

Set hashie logger to Rails logger or disable it entirely. This feels a little aggressive, though. e.g. https://github.com/berkshelf/berkshelf/pull/1668/files

**Comment by maxkwallace** [comment]
@bentheax thanks for reporting! I'm seeing this too.

**Comment by bquorning** [comment]
Fixed in [v1.14.3](https://github.com/zendesk/zendesk_api_client_rb/releases/tag/v1.14.3) by #327.
